stylix/palette.nix

```diff
@@ -10,21 +10,51 @@ let
   opts = options.stylix;
 in
 {
+  imports = lib.singleton (
+    lib.mkRenamedOptionModule
+      [ "stylix" "polarity" ]
+      [ "stylix" "colorGeneration" "polarity" ]
+  );
   options.stylix = {
-    polarity = lib.mkOption {
-      type = lib.types.enum [
-        "either"
-        "light"
-        "dark"
-      ];
-      default = "either";
-      description = ''
-        Use this option to force a light or dark theme.
+    colorGeneration = {
+      scheme = lib.mkOption {
+        type = lib.types.enum [
+          "content"
+          "expressive"
+          "fidelity"
+          "fruit-salad"
+          "monochrome"
+          "neutral"
+          "rainbow"
+          "tonal-spot"
+          "vibrant"
+        ];
+        default = "content";
+        description = ''
+          [Matugen](https://github.com/InioX/matugen)'s color scheme type.
+        '';
+        apply = value: "scheme-${value}";
+      };
+      contrast = lib.mkOption {
+        type = lib.types.addCheck lib.types.float (x: x >= -1.0 && x <= 1.0);
+        default = 0.0;
+        description = ''
+          Use this option to change the generated color scheme's contrast.
 
-        By default we will select whichever is ranked better by the genetic
-        algorithm. This aims to get good contrast between the foreground and
-        background, as well as some variety in the highlight colours.
-      '';
+          `-1` represents minimum contrast, `0` represents standard (i.e. the
+          design as spec'd), and `1` represents maximum contrast.
+        '';
+      };
```

Comment:
The `stylix.colorGeneration.scheme` and `stylix.colorGeneration.contrast` options should be documented. Probably in the following section:

https://github.com/nix-community/stylix/blob/182be7885cf170dc0dbff759a55f5c71c2253e56/doc/src/configuration.md?plain=1#L73-L110